runtime/interpreter/encode.go

```diff
@@ -0,0 +1,899 @@
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareInt32(v Int32Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagInt32Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareInt64(v Int64Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagInt64Value,
+		Content: v,
+	}
+}
+func (e *Encoder) prepareInt128(v Int128Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagInt128Value,
+		Content: e.prepareBig(v.BigInt),
+	}
+}
+
+func (e *Encoder) prepareInt256(v Int256Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagInt256Value,
+		Content: e.prepareBig(v.BigInt),
+	}
+}
+
+func (e *Encoder) prepareUInt(v UIntValue) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUIntValue,
+		Content: e.prepareBig(v.BigInt),
+	}
+}
+
+func (e *Encoder) prepareUInt8(v UInt8Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt8Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareUInt16(v UInt16Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt16Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareUInt32(v UInt32Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt32Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareUInt64(v UInt64Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt64Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareUInt128(v UInt128Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt128Value,
+		Content: e.prepareBig(v.BigInt),
+	}
+}
+
+func (e *Encoder) prepareUInt256(v UInt256Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUInt256Value,
+		Content: e.prepareBig(v.BigInt),
+	}
+}
+
+func (e *Encoder) prepareWord8(v Word8Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagWord8Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareWord16(v Word16Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagWord16Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareWord32(v Word32Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagWord32Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareWord64(v Word64Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagWord64Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareFix64(v Fix64Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagFix64Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareUFix64(v UFix64Value) cbor.Tag {
+	return cbor.Tag{
+		Number:  cborTagUFix64Value,
+		Content: v,
+	}
+}
+
+func (e *Encoder) prepareString(v *StringValue) string {
+	return v.Str
+}
+
+func (e *Encoder) prepareArray(v *ArrayValue) ([]interface{}, error) {
+	result := make([]interface{}, len(v.Values))
+
+	for i, value := range v.Values {
+		prepared, err := e.prepare(value)
+		if err != nil {
+			return nil, err
+		}
+		result[i] = prepared
+	}
+
+	return result, nil
+}
+
+type encodedDictionaryValue struct {
+	Keys    []interface{}          `cbor:"0,keyasint"`
+	Entries map[string]interface{} `cbor:"1,keyasint"`
+}
+
+// TODO: optimize: use CBOR map, but unclear how to preserve ordering
```

Comment:
I think you might be able to use `SortBytewiseLexical ` from https://github.com/fxamacker/cbor#encoding-options-1

Reply:
Ah, I meant preserving the insertion order we keep for the keys of the Cadence dictionary value. 

We could use the byteweise lexical ordering also for the Cadence value 🤔